<launch>
  <node pkg="xqserial_server" type="xqserial_server" name="motor_driver" respawn="true" respawn_delay="10" output="screen">
    <param name="debug_flag" value="false"/>
    <param name="wheel_separation" value="0.40"/>
    <param name="wheel_radius" value="0.085"/>
    <param name="max_speed" value="3.0"/>
    <param name="port_car" value="/dev/ttyS0"/>
    <param name="port_imu" value="/dev/STM32_UART"/>
    <param name="power_scale" value="1.0155"/>
    <param name="tran_dist" value="0.3"/>
    <param name="angle_limit" value="1.6"/>
    <param name="x_limit" value="1.5"/>
    <param name="y_limit" value="0.27"/>
    <param name="r_min" value="0.7"/>
    <param name="cmd_topic" value="/cmd_vel" />
    <remap from="/scan" to="/scan_filtered" />
  </node>
  <node pkg="tf" type="static_transform_publisher" name="baselink_broadcaster" args="0 0 0.15 0 0 0 1 base_footprint base_link 50">
  </node>
  <node pkg="tf" type="static_transform_publisher" name="baselinkSonar_broadcaster1" args="0.1 -0.150 0.44 0 0 0  base_link sonar1 50"/>
  <node pkg="tf" type="static_transform_publisher" name="baselinkSonar_broadcaster2" args="0.1 0.150 0.44 0 0 0  base_link sonar2 50"/>
  <node pkg="tf" type="static_transform_publisher" name="baselinkSonar_broadcaster3" args="0.1 -0.150 0.64 0 0 0  base_link sonar1 50"/>
  <node pkg="tf" type="static_transform_publisher" name="baselinkSonar_broadcaster4" args="0.1 0.150 0.64 0 0 0  base_link sonar2 50"/>
  <node pkg="tf" type="static_transform_publisher" name="imulink_broadcaster" args="0.07 -0.03 0.2 0 0 0 1 base_link imu 50"/>

  <node pkg="self_checking" type="SELF_CHECK_NODE" name="SELF_CHECK_NODE"/>

</launch>
